ascii.py

```diff
@@ -0,0 +1,31 @@
+class Ascii():
+
+	def __init__(self):
+		pass
+
+	def draw_square(self, piece):
+		space = "| " + piece + " "
+		return space
+
+	def draw_row(self, row_marker, pieces):
+		bottom_half = "|\n    " + "|____" * 8 + "|"
+		row = " " + row_marker + "  "
+		for piece in pieces:
+			row += (self.draw_square(piece))
+		row += bottom_half
+		print(row)
```

Comment:
In general, it is hard to test things that combine the logic of what to do with some side-effect.

It will be easier to reason about this, and to write unit tests, if these methods simply calculate and build up a string to return. Then callers can print it out when and where they need to. And unit tests can simply get back a nice string and compare it to expected results.

This will make writing a unit test easy, since you can use this loigic

Reply:
I think I understand. It's probably several steps down the road, but I don't know how to create unit tests in python, so I'll definitely want some input on that eventually!